soapui/src/main/java/com/eviware/soapui/impl/rest/actions/method/SaveRequestAction.java

```diff
@@ -67,9 +67,11 @@ public void showNewRestRequestDialog() {
         if (dialog.show()) {
             String requestName = dialog.getValue(SaveRequestAction.Form.RESOURCENAME);
             RestRequest request = addRequest(context, requestName);
-            UISupport.invokeLater(() -> SoapUI.getNavigator().updateUI());
             if (dialog.getBooleanValue(SaveRequestAction.Form.OPENSREQUEST)) {
                 UISupport.selectAndShow(request);
+            } else {
+                //SOAPUIOS-447
+                UISupport.select(SoapUI.getNavigator().getSelectedItem());

```

Comment:
getSelectedItem() can potentially return null. Will it be fine or will it never happen in this case?

Reply:
It's a null-safe method